src/relativePxth.ts

```diff
@@ -0,0 +1,60 @@
+import invariant from 'tiny-invariant';
+
+import { createPxth } from './createPxth';
+import { getPxthSegments } from './getPxthSegments';
+import { Pxth } from './Pxth';
+import { pxthToString } from './pxthToString';
+import { samePxth } from './samePxth';
+
+/**
+ * Returns relative path. If subPath is not child of basePath, it will throw an error.
+ * @example
+ * relativePxth(['hello', 'world'], ['hello', 'world', 'asdf']) -> ['asdf'],
+ * relativePxth(['a', 'b', 'c'], ['a', 'b', 'c', 'd', 'e']) -> ['d', 'e'],
+ * relativePxth(['a', 'b', 'c'], ['a', 'b', 'c']) -> ROOT_PATH,
+ * relativePxth(['a'], ['b']) -> Error
+ */
+export const relativePxth = <A, B>(
+    basePath: Pxth<A>,
+    subPath: Pxth<B>,
+): Pxth<B> => {
+    const basePathSegments = getPxthSegments(basePath);
+    const subPathSegments = getPxthSegments(subPath);
+
+    if (basePathSegments.length === 0 && subPathSegments.length === 0) {
+        return createPxth([]);
+    }
+
+    if (basePathSegments.length === 0) {
+        return subPath;
+    }
+
+    invariant(
+        subPathSegments.length !== 0,
+        'ROOT_PATH symbol cannot be sub path of any path',
+    );
+
+    invariant(
+        subPathSegments.length >= basePathSegments.length,
+        `Sub path (${String(
+            pxthToString(subPath),
+        )}) must be longer than base path (${String(pxthToString(basePath))})`,
+    );

```

Comment:
No need for special check on path lengths - it should be handled by isInnerPath function